### ISSUE TITLE
Receiving Tax and Profit

### ISSUE THREAD

**Issue by bkrajendra** [jekkos/opensourcepos#502]
Hi friends, Im having little confusing conditions for me. 

1) In my case my vendor charges me for tax (5%). So if I receive a product of cost 48.63 INR (Cost Price ), vendor applies tax on it and adds the tax to total invoice.

2) I've all my products marked as inclusive of all taxes and sell it at 60 INR (Retail Price). 
Now ive setup this tax rates in my Items and set store config as "Tax Included" so that my final cost for my customer should not go ahead of 60/-

3) When Im running Detail Report, I'm getting profit calculated as 11.37/- 
by considering my situation i need the 5% tax (2.86/-) has to be subtracted from profit, which is not happening here. so my final profit should be 8.51/-

4) Also I want to generate a receiving invoice which should consists of Item Cost+tax.

I'm not expert in financials. need guidance on how it should be done. I'm not expecting code, but brief direction about is it possible and how it can be achieved in existing OSPOS.

@pappastech 

![report](https://cloud.githubusercontent.com/assets/994083/14767582/c8979cfc-0a46-11e6-9047-cfc803090d95.PNG)

**Comment by RamkrishnaMondal** [comment]
For 1,2 & 3 ---
Do not  use Tax-included, it will increase the profit by tax percentage.
Rs.48.63 is your 'Basic purchase price' , your cost price is Rs. 51.0615 say Rs.51.06.
Without using  tax included: 
![screenshot from 2016-04-24 18 57 58](https://cloud.githubusercontent.com/assets/14151032/14767931/3913e9f0-0a50-11e6-9bc3-619d3c57b41f.png)

With tax-included:
![screenshot from 2016-04-24 19 11 09](https://cloud.githubusercontent.com/assets/14151032/14767936/54ab3664-0a50-11e6-8167-d99393770d43.png)

**Comment by osworx** [comment]
Basically you should learn how business is done.
Does no matter which tax your vendor is calculating, your price per item is the NET price - the tax is only for end-consumers, never for B2B (I assume that you run your store as business and not as private).
Further, the tax you got from your vendor decrease the overall tax value you have to send to your tax authority.

For example: 
You buy 1 item net value 50,- /  tax of 5% = 5 => gross value = 52,50 (you now have a credit of 2,50 against your tax authority).
You now sell this item for 63,-, means you have added 20% margin and your local tax of 3% from 60 (50 + 20%) + 3%.
You now have to pay 0,50 (3 - 2,50) to your financial authority.
Question: what is your profit now?

**Comment by bkrajendra** [comment]
@RamkrishnaMondal , thank you for quick reply.
@osworx .. Thank you for educating me. I need it a lot! 
But I'm still super confused!
1) @RamkrishnaMondal  - I did get what you said. 
But When I get Invoice from my vendor i get it as :

![rep](https://cloud.githubusercontent.com/assets/994083/14769901/db537976-0a80-11e6-89ee-7b6082e24469.PNG)

Now which cost should i enter while receiving?
Rate (48.63) or 48.63+tax ?

What should be MRP (my Selling price). 60 or 57/- ?

what setting should be applied to Tax Included?

How should I calculate profit in this case.?
From accounting and Indian tax point of view, what should be standard practice.
Do I need to pay tax to authorities?
As I already said all my items are marked as Incl of all taxes. 
So my all invoice (Receipts) to my customers should not show any tax.
(Sorry for too many accounting questions. But I'm a programmer with very average knowledge of finance. I'm into electronics parts and embedded software design.)

**Comment by daN4cat** [comment]
Besides all the accounting part (normally is net cost, sell price with or without tax depends on where you live but you have taxes anyway), to have 48.63 at cost price and 60 at retail price is really a slim margin.
Before entering a business consider your margins! You should have at least 40%.

**Comment by RamkrishnaMondal** [comment]
@bkrajendra 
1. Un check 'Tax-included' in general config.
2. Enter 48.63 as cost while receiving.
3. Enter (60/1.05) = 57.14 as Retail Price. 
4. Un check 'Show Tax' on Receipt Config , if you do not wish to show tax on sale receipts.
5. If your sales value exceeds Rs 5 lacs pa (Rs 10 lacs in some states) , you have to apply for VAT        registration .
Hope this helps.

@daN4cat  40% is more than a dream margin in India. People even trade @ 1% - 1.5% margin. But  it's a country of 125 Cr people...... you can survive by selling in bulk quantity.

**Comment by bkrajendra** [comment]
@RamkrishnaMondal  the only problem with your approach is that the generated receipt is misleading.
![rec](https://cloud.githubusercontent.com/assets/994083/14773729/7515e978-0acb-11e6-9b63-3382372e2d52.PNG)

it shows product cost as 57.14 and displays 60in total.
my all products clearly says incl of all taxes. so its necessary to display receipt with 60/- only (product MRP).

I think I first need to learn about my own business and then come here with more questions.
I'll post with more explanation here later in afternoon.

**Comment by RamkrishnaMondal** [comment]
I can understand your problem, as you can not show tax on receipt separately, the only option you have is to use tax included.  In that case your cost will be 51.06(basic rate+portion of input tax on it) and retail price will be 60, though your profit will be 5% higher than actual profit due to bug in OSPOS.

**Comment by daN4cat** [comment]
I will check this point because if a taxation is declared it should not be part of the profit.
Of course there is not that much we can do for the tax on source (cost), although one could argue that would be worth adding so you exactly know what is your profit because the tax at source is deducted from the tax to customer pays.

By the way, I don't see why a receipt should not contain the tax %, what is the problem with that to be shown to the customers? In the end it's public knowledge and you show that you charge and pay.

**Comment by bkrajendra** [comment]
@daN4cat you are correct, there is need to have source tax to be available in receiving. This will make it possible to generate accurate report for "Suppliers Summary Report" which will match with supplier Invoice. 

I'm trying to understand the structure of OSPOS to alter the reporting and database management. 
@RamkrishnaMondal yes thats the only thing i can do now. But now how to make cost = basic rate+input tax while receiving so that it will be easier to manage purchase and profit.

What if i add two columns to table ospos_receivings_items
item_base_cost and item_input_tax
And then instead of taking item_cost_price from user it will be calculated from
item_cost_price  = (item_base_cost x item_input_tax/100)+item_base_cost

by this I will preserve the suppliers costing and input tax and report can be generated as per my requirement as shown earlier.

**Comment by RamkrishnaMondal** [comment]
Sometimes, I really feel the need of  "Takings" and "Edit/Update" buttons in Receivings.

**Comment by jekkos** [comment]
Receivings can be edited through detailed receivings reports, that's the only way currently

**Comment by RamkrishnaMondal** [comment]
@jekkos  Please consider adding a TAKINGS button.

**Comment by bkrajendra** [comment]
Hi @RamkrishnaMondal  or @jekkos can you explain me how can I just Put 2 text boxes for Purchase Tax and Purchase Price on receiving form. I want these boxes also to get this details from database table for each product and also to save these value back. Cost should be calculated from purchase price and purchase tax.

I just want to know data save/retrieve mechanism used in receiving controller-view-model

I found codeigniter toooo much difficult to understand. Im very well conversant with Laravel framework and its simplicity. Codeigniter confuses me a lot. Specially this OSPOS structure is too complicated.

Either explain me how can I do it or just point me to some resources.
Thanks in advance.!!!

**Comment by RamkrishnaMondal** [comment]
Codeigniter is very well documented : http://www.codeigniter.com/user_guide/

Video tutorial in hindi : https://www.youtube.com/watch?v=wRmBCh1J464&list=PLHFGzOr0F8DI5IBxWWzHvdUL_HmWUAqg2
